From 1207becd01730ea1083722bdfc26e45d29dc202a Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Fri, 19 Oct 2018 11:56:11 +0200
Subject: [PATCH 4/4] Merge android-9.0.0_r12

commit f9c275979dc3beac879a6123e733c9220a701a5e
Merge: ae9650a6 a5471138
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Sun Jul 1 03:09:03 2018 +0000

    Snap for 4872253 from a5471138c51014e1cfa84a50bf3c7da26e5a06cb to pi-dr1-release

    Change-Id: I1d04a249bee40c2a22abc1b3794d5781225d5397

commit a5471138c51014e1cfa84a50bf3c7da26e5a06cb
Merge: 9747eed8 40c869b0
Author: TreeHugger Robot <treehugger-gerrit@google.com>
Date:   Sat Jun 30 00:46:02 2018 +0000

    Merge "Remove workaround for cortex-a55/a75" into pi-dev

commit ae9650a650d608f0a7225c15f77dd4c8ee1eb72a
Merge: 4e17cb72 39299e61
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Jun 26 03:14:40 2018 +0000

    Snap for 4861493 from 39299e61610198b3d3c511a58ea540471d8e29c2 to pi-dr1-release

    Change-Id: Id7ffc9cc7cabc0fcb9be4ec77897f8b2a4119e3f

commit 4e17cb728c01125a535d6f3fd2bde75598a7b3eb
Merge: c6262e42 1c1c5602
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Sat Jun 23 18:20:17 2018 +0000

    Merge cherrypicks of [4410329, 4410313, 4410314, 4409471] into pi-dr1-release

    Change-Id: I6e699595f29ca170fdca65b0af40cbb3d0b57a12

commit 1c1c56023cf29e5e59020c161ab0818f3744c7ce
Author: Yi Kong <yikong@google.com>
Date:   Sun Jun 17 02:15:24 2018 -0700

    Remove workaround for cortex-a55/a75

    Test: m checkbuild
    Test: boot on a55/a75 device, pass bionic tests
    Bug: 110235326
    Change-Id: I5ab2102352a6efe1173b3097875e6e779d4a1a09
    Merged-In: I5ab2102352a6efe1173b3097875e6e779d4a1a09
    (cherry picked from commit 9a350e644b4e5762df5838e007a55af68449c937)
    (cherry picked from commit 40c869b0aeff3df0d34a6b0540c1a04c142552ff)

commit d7f56e023ca12ed031c609df76e969422a70dc8b
Author: Yi Kong <yikong@google.com>
Date:   Thu Jun 14 22:38:10 2018 -0700

    Update ToolingCFlags overrides to include other new architectures

    Bug: 110235326
    Test: m checkbuild
    Change-Id: Ifaa35db08d35ed3cb14fce3e9c5643f26bc3f706
    Merged-In: I9d0ada05d95bb260500c1d694332a73363b0f299
    (cherry picked from commit 32a779171b78615b231d0680872942d2dd07b519)
    (cherry picked from commit 9747eed816fda6b1766f20dfb84b43ffe4abc1be)

commit 40c869b0aeff3df0d34a6b0540c1a04c142552ff
Author: Yi Kong <yikong@google.com>
Date:   Sun Jun 17 02:15:24 2018 -0700

    Remove workaround for cortex-a55/a75

    Test: m checkbuild
    Test: boot on a55/a75 device, pass bionic tests
    Bug: 110235326
    Change-Id: I5ab2102352a6efe1173b3097875e6e779d4a1a09
    Merged-In: I5ab2102352a6efe1173b3097875e6e779d4a1a09
    (cherry picked from commit 9a350e644b4e5762df5838e007a55af68449c937)

commit 9747eed816fda6b1766f20dfb84b43ffe4abc1be
Author: Yi Kong <yikong@google.com>
Date:   Thu Jun 14 22:38:10 2018 -0700

    Update ToolingCFlags overrides to include other new architectures

    Bug: 110235326
    Test: m checkbuild
    Change-Id: Ifaa35db08d35ed3cb14fce3e9c5643f26bc3f706
    Merged-In: I9d0ada05d95bb260500c1d694332a73363b0f299
    (cherry picked from commit 32a779171b78615b231d0680872942d2dd07b519)

commit c6262e422da2d8dd5a0a73865c33de28cf3ce9ba
Merge: 2d5db9f3 39299e61
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Thu Jun 14 03:02:55 2018 +0000

    Snap for 4839323 from 39299e61610198b3d3c511a58ea540471d8e29c2 to pi-dr1-release

    Change-Id: I8bd1cec284d63a3f7c8be47fa440b4e493a71c2c

commit 39299e61610198b3d3c511a58ea540471d8e29c2
Author: Logan Chien <loganchien@google.com>
Date:   Wed Jun 13 22:32:16 2018 +0800

    Fix VNDK-Ext ABI check regression

    VNDK-Ext are modules with `vndk.enabled: true` but not having
    `vendor_available: true`.  In addition, VNDK-Ext should be checked by
    source ABI checker.  This change fixes the regression introduced in

    Bug: 110142940
    Test: Create libminijail_ext, break some ABIs, and see an error.
    Merged-In: I8b47ac12d2e132f641129c9549ed22c3971d6c89
    Change-Id: I8b47ac12d2e132f641129c9549ed22c3971d6c89
    (cherry picked from commit ef1ff3de9804d8e99df06f0874c974b3f1e85e36)
    Signed-off-by: Jayant Chowdhary <jchowdhary@google.com>

commit 2d5db9f33e2b97deb0abc0405ea3adfbee28ec5d
Merge: 9ccbba02 19b17426
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Sun Jun 10 03:07:09 2018 +0000

    Snap for 4832091 from 19b1742614570b0debcdd13238917126924d321c to pi-dr1-release

    Change-Id: Ibb0425679a79a6aa47c896ce177ae1884a92fde4

Change-Id: Ic4e2011f9a665ad149bf7e008068ebdc07003757
---
 cc/config/arm64_device.go | 9 +++------
 cc/config/arm_device.go   | 4 ++--
 cc/sabi.go                | 9 +++++++--
 3 files changed, 12 insertions(+), 10 deletions(-)

diff --git a/cc/config/arm64_device.go b/cc/config/arm64_device.go
index 11625e80..8be33a85 100644
--- a/cc/config/arm64_device.go
+++ b/cc/config/arm64_device.go
@@ -50,15 +50,12 @@ var (
 			"-mcpu=cortex-a53",
 		},
 		"cortex-a55": []string{
-			// The cortex-a55 target is not yet supported,
-			// so use cortex-a53.
-			"-mcpu=cortex-a53",
+			"-mcpu=cortex-a55",
 		},
 		"cortex-a75": []string{
-			// Use the cortex-a53 since it is similar to the little
+			// Use the cortex-a55 since it is similar to the little
 			// core (cortex-a55) and is sensitive to ordering.
-			// The cortex-a55 target is not yet supported.
-			"-mcpu=cortex-a53",
+			"-mcpu=cortex-a55",
 		},
 		"kryo": []string{
 			// Use the cortex-a57 cpu since some compilers
diff --git a/cc/config/arm_device.go b/cc/config/arm_device.go
index 4e0f6e05..cad1b161 100644
--- a/cc/config/arm_device.go
+++ b/cc/config/arm_device.go
@@ -98,7 +98,7 @@ var (
 			"-D__ARM_FEATURE_LPAE=1",
 		},
 		"cortex-a55": []string{
-			"-mcpu=cortex-a53",
+			"-mcpu=cortex-a55",
 			"-mfpu=neon-fp-armv8",
 			// Fake an ARM compiler flag as these processors support LPAE which GCC/clang
 			// don't advertise.
@@ -107,7 +107,7 @@ var (
 			"-D__ARM_FEATURE_LPAE=1",
 		},
 		"cortex-a75": []string{
-			"-mcpu=cortex-a53",
+			"-mcpu=cortex-a55",
 			"-mfpu=neon-fp-armv8",
 			// Fake an ARM compiler flag as these processors support LPAE which GCC/clang
 			// don't advertise.
diff --git a/cc/sabi.go b/cc/sabi.go
index f5a7c774..42b2f352 100644
--- a/cc/sabi.go
+++ b/cc/sabi.go
@@ -74,8 +74,13 @@ func (sabimod *sabi) flags(ctx ModuleContext, flags Flags) Flags {
 
 	// RSClang does not support recent mcpu option likes exynos-m2.
 	// So we need overriding mcpu option when we want to use it.
-	if ctx.Arch().CpuVariant == "exynos-m2" {
-		flags.ToolingCFlags = append(flags.ToolingCFlags, "-mcpu=cortex-a53")
+	mappedArch := map[string]string{
+		"exynos-m2":  "cortex-a53",
+		"cortex-a55": "cortex-a53",
+		"cortex-a75": "cortex-a57",
+	}
+	if arch, ok := mappedArch[ctx.Arch().CpuVariant]; ok {
+		flags.ToolingCFlags = append(flags.ToolingCFlags, "-mcpu="+arch)
 	}
 
 	return flags
-- 
2.17.1

